components/admin_console/custom_plugin_settings/index.js

```diff
@@ -27,13 +31,35 @@ function makeGetPluginSchema() {
             let settings = [];
             if (plugin.settings_schema && plugin.settings_schema.settings) {
                 settings = plugin.settings_schema.settings.map((setting) => {
+                    const key = setting.key.toLowerCase();
+                    let component = null;
+                    let bannerType = '';
+                    let type = setting.type;
+                    let displayName = setting.display_name;
+                    let isDisabled = it.stateIsFalse(pluginEnabledConfigKey);
+
+                    if (customComponents[key]) {
```

Comment:
Shouldn't it match only if the keys match and the type is CUSTOM?, I mean, if the keys matches and I pass another type, is going to work?

Reply:
It takes the custom component registered under that settings key regardless of the type specified and overrides the type to be custom [inside that if](https://github.com/mattermost/mattermost-webapp/blob/32a8d42aa13de6171340cb9dad1d575512f8bccc/components/admin_console/custom_plugin_settings/index.js#L44). The assumption is that the act of registration trumps over the setting type in the manifest file.

Reply:
Would it make sense to throw an error if we try to register a custom component that isn't marked as `TYPE_CUSTOM` to avoid weirdness where you can use a default editor when disabled, and the custom component when enabled?

Reply:
When it is disabled you still have no way of knowing if that setting is supposed to be registered (`customComponents[key]` is always empty until enabled) and it will still display the default editor if misconfigured. 

The error message will show only if it's enabled and the type is not custom. At this point it might as well to override to the custom type as it does now.

Thoughts?